### PR TITLE
Add `objectscript.explorer.alwaysShowServerCopy` setting

### LINKED ISSUE #494: Explorer view should always show what is on the server
Relocating here from https://github.com/intersystems/language-server/issues/78

When browsing the class tree from the Objectscript Explorer VS Code will always show the local exported copy of the file. This leaves no way to see what is actually on the server.  I expected the explorer to be analogous to the Atelier server tab.

Steps to reproduce:

1. export a file
2. modify the file outside VS code (e.g. a `git pull` or fake it using `vi`)
3. browse to the file in the explorer, notice the tooltip is pointing to the working copy and the change is show even though it was never imported and compiled.

### PR DESCRIPTION
**Pull request by isc-bsaviano**
This PR fixes #494 

I verified that the explorer works properly with the new setting enabled or disabled, and that the LS's go to definition provider still opens the local copy of the file regardless of the new setting's value.

### CODE REVIEW

File: docs/SettingsReference.md [hunk 1]
```diff
@@ -41,6 +41,7 @@ The extensions in the InterSystems ObjectScript Extension Pack provide many sett
 | `"objectscript.compileOnSave"` | Automatically compile an InterSystems file when saved in the editor. | `boolean` | `true` | |
 | `"objectscript.conn"` | Configures the active server connection. | `object` | `undefined` | See the [Configuration page](../configuration/#config-server-conn) for more details on configuring server connections. |
 | `"objectscript.debug.debugThisMethod"` | Show inline `Debug this method` CodeLens action for ClassMethods. | `boolean` | `true` | |
+| `"objectscript.explorer.alwaysShowServerCopy"` | Always show the server copy of a document in the ObjectScript Explorer, like the Atelier Server Explorer tab. | `boolean` | `false` | |
```
Comment (gjsjohnmurray) [inline docs/SettingsReference.md:44]: I suggest dropping the mention of Atelier.

File: package.json [hunk 2]
```diff
@@ -1072,6 +1072,11 @@
           "description": "Automatically collapse all folding ranges when a class is opened for the first time.",
           "type": "boolean",
           "default": false
+        },
+        "objectscript.explorer.alwaysShowServerCopy": {
+          "description": "Always show the server copy of a document in the ObjectScript Explorer, like the Atelier Server Explorer tab.",
```
Comment (gjsjohnmurray) [inline package.json:1077]: I suggest dropping the mention of Atelier.

File: src/explorer/models/classNode.ts [hunk 3]
```diff
@@ -12,15 +13,24 @@ export class ClassNode extends NodeBase {
...
-        arguments: [itemUri],
+        arguments: [isLocalFile && !showServerCopy ? itemUri : serverCopyUri],
         command: "vscode-objectscript.explorer.open",
         title: "Open Class",
       },
-      resourceUri: isLocalFile ? itemUri : undefined,
+      resourceUri: isLocalFile && !showServerCopy ? itemUri : undefined,
       contextValue: "dataNode:classNode",
       label: `${displayName}`,
       tooltip: isLocalFile ? undefined : this.fullName,
```
Comment (gjsjohnmurray) [inline src/explorer/models/classNode.ts:36]: Should this also take account of the new setting?

File: src/explorer/models/cspFileNode.ts [hunk 4]
```diff
@@ -12,15 +13,24 @@ export class CSPFileNode extends NodeBase {
...
-        arguments: [itemUri],
+        arguments: [isLocalFile && !showServerCopy ? itemUri : serverCopyUri],
         command: "vscode-objectscript.explorer.open",
         title: "Open File",
       },
-      resourceUri: isLocalFile ? itemUri : undefined,
+      resourceUri: isLocalFile && !showServerCopy ? itemUri : undefined,
       contextValue: CSPFileNode.contextValue,
       label: `${displayName}`,
       tooltip: isLocalFile ? undefined : this.fullName,
```
Comment (gjsjohnmurray) [inline src/explorer/models/cspFileNode.ts:36]: Should this also take account of the new setting?

File: src/explorer/models/routineNode.ts [hunk 5]
```diff
@@ -12,15 +13,24 @@ export class RoutineNode extends NodeBase {
...
-        arguments: [itemUri],
+        arguments: [isLocalFile && !showServerCopy ? itemUri : serverCopyUri],
         command: "vscode-objectscript.explorer.open",
         title: "Open Routine",
       },
-      resourceUri: isLocalFile ? itemUri : undefined,
+      resourceUri: isLocalFile && !showServerCopy ? itemUri : undefined,
       contextValue: "dataNode:routineNode",
       label: `${displayName}`,
       tooltip: isLocalFile ? undefined : this.fullName,
```
Comment (gjsjohnmurray) [inline src/explorer/models/routineNode.ts:36]: Should this also take account of the new setting?

**Review by gjsjohnmurray — CHANGES_REQUESTED** [review]

File: src/explorer/models/classNode.ts (hunk 3 above)
Comment (isc-bsaviano) [reply src/explorer/models/classNode.ts:36]: The current behavior is to only attach a resource URI when the file is found locally. If we want to change that I think it should be a separate PR.

Comment (gjsjohnmurray) [reply src/explorer/models/classNode.ts:36]: I was commenting on the line that conditionally sets tooltip

Comment (isc-bsaviano) [reply src/explorer/models/classNode.ts:36]: Sorry, I misread that. Yes, I'll fix that.

**Review by gjsjohnmurray — APPROVED** [review]
LGTM but I haven't tested it myself.

**Comment by isc-bsaviano** [comment]
@gjsjohnmurray I'll merge it once the issue reporter confirms that it works for him

**Review by daimor — APPROVED** [review]

**Pull request merged**